### DISCUSSION TITLE
[2024-12-09] Libs and Engine Team Demo Day

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#11806] [Weekly Demo Day]
- James: Conditionally trigger AMI build. #11680
- James: Fix for Data Quality perf and issue other tweaks. #11711
- James: Fast Excel Reading. #10877
- Radosław: Finishing #11546 
- Radosław: Refactor common SPI logic, handle SPI registrations with missing Enso library #11722
- Radosław: Introduce `Value_Type.Null`, fix edge cases for `read_many` #11737
- Radosław: Fix `Data.list` for other file systems #11743
- Radosław: Bump required launcher version for new engine releases, forcing an upgrade to compatible version #11796
- Radosław: Allow `Data.list` to cross data links #11802
- Greg: Clear cache on reload #11673
- Greg: Reissue request when cached tempfile is missing #11689
- Greg: Fix HTTP cache assertion failure for small disks #11738
- Greg: Sort output of `File.list` #11752
- Ad: SQLServer Aggregate #11811 
- Ad: Fix Sorting in Snowflake #11636
- Ad: Start offset design
- Martha: Add date support for filter drilldown #11626
- Martha: Drive data quality metrics from enso code #11638
- Martha: Filter drilldown for mixed column #11694
- Martha: Wip/mk/display message on warning #11705
- Martha: check for long negavtive numbers not just positive #11750
- Martha: Row display in Table viz in correct order #11764
- Martha: Correctly format big integers #11799
- Hubert: Delegate slf4j in HostClassLoader to system class loader #11670
- Hubert: Simplify shutdown logic on client disconnect in project-manager #11712
- Pavel: Builtins expose Enso methods #11687
- Pavel: Convert rest of global typing pass group to mini passes #11717
- Pavel: Refactor runtime-integration-tests - move some tests to more dedicated packages #11720
- Pavel: [Poc] Generate IR definitions - superclass approach #11770
- Dmitry: Update licenses codeowners #11663
- Dmitry: Fix RuntimeAsyncCommandsTest #11668
- Dmitry: Fix std-benchmarks module path #11674
- Dmitry: Fix scheduled nightly release #11675
- Dmitry: Use GitHub concurrency setting #11704
- Dmitry: Fix inputs for scheduled Nightly workflow #11708
- Dmitry: Run workflows depending on changed files #11723
- Dmitry: Fix nightly build #11726
- Dmitry: Compute correct changeset when swapping nodes #11765
- Dmitry: Schedule rename symbol job after compilation #11780
- Dmitry: Project Manager fails to create directory when one exists #11804

**Comment by JaroslavTulach** [comment]
## Debugging Standard libraries 
 
- #11736

![NetBeans+Enso debugging](https://github.com/user-attachments/assets/c96d2fae-8680-4317-8dbf-a75bc3a295f0)

- the `distribution/lib/Standard` versions are loaded
- they have the Git history associated
- editing them will update the "right" sources
- running `runBuildEngineDistribution` updates the ` build-distribution` as usual
- the same support works in VSCode

![VSCode+Enso debugging](https://github.com/user-attachments/assets/3a965e40-63c8-4e7e-985c-6ac4ca24e48e)

- also works in Chrome Dev tools

![ChromeDev tools debugging](https://github.com/user-attachments/assets/0b1d0389-a527-44d5-8afe-d59a5ed9d8b0)

## No More of Forgetting `Error`

- No longer longer **forgetting** `DataflowError`
- #11777
- [six bugs](https://github.com/enso-org/enso/pull/11777#issuecomment-2527968617) revealed in `test/*_Tests`


## Docs

> [!WARNING]
> Finally **updated** (a bit of) the outdated **documentation** 

- Starting to [read from here](https://github.com/enso-org/enso/blob/81b0fcc3ef3f80323f7de357c60bcea2185c1ad7/docs/types/README.md)!
- #11775

## Miscellaneous

- Avoid exposing EnsoMultiValue getters #11642
- Robust against `IllegalArgumentException` #11662
- Better Module_Does_Not_Exist exception #11713
- Avoid storing 0.0.0-dev into package.yaml #11805
